eZ/Publish/Core/Repository/ContentTypeService.php

```diff
@@ -1639,4 +1640,13 @@ public function removeContentTypeTranslation(APIContentTypeDraft $contentTypeDra
 
         return $this->contentTypeDomainMapper->buildContentTypeDraftDomainObject($contentType);
     }
+
+    public function deleteUserDrafts(int $userId): void
+    {
+        if ($this->repository->getPermissionResolver()->hasAccess('class', 'delete') !== true) {
```

Comment:
This is only optional half of the remark I made
> It's not safe to just delete regardless of system state.
> You should check first if user does not exist and proceed only if it does.

> Optional: If it does exist, then at least class/delete should be checked.

So it's good, but only if user exists. If it does not, it's a simple cleanup operation, which will fail in case user has `user/delete` permission, but don't have `class/delete`.